### PR TITLE
Support firmware upgrades

### PR DESCRIPTION
**Pull request by davidyscott**
The primary task was ICST-1132, to add support for the Firmware Update APIs.  In writing tests for that code, I found some things I thought could be improved, and I felt like our public client libraries, especially the Java ones, should be the best we can make them.

Brief summary of overarching changes:
 - Remove "created by jcombopiano" file header (this is the primary reason so many files were touched)
 - Optimize all imports (there were unused imports in a lot of files, so I figured I may as well go for completeness... again, sorry for touching all the files --- isolating that to its own commit wasn't really good enough, I should have made that a separate PR)
 - Centralize a bunch of boilerplate test code into a new TestUtils class
 - Remove the one-off implementation that was used to enable de/serialization of specifically java.time.ZonedDateTime in both Gson (tests) and Jackson and replace it with properly configuring and using open-source libraries for full handling of the java.time package

New overarching changes:
 - Use BASE_PATH constant in most Controllers
 - Refactor the signatures of RestApiHelper methods to use varargs for path segments

### CODE REVIEW

File: src/main/java/com/impinj/itemsense/client/coordinator/currentZoneMap/CurrentZoneMapController.java [hunk 1]
```diff
@@ -16,11 +13,11 @@
 
     public CurrentZoneMapController(WebTarget target) {
         this.target = target;
-        this.restApiHelper = new RestApiHelper<CurrentZoneMap>(CurrentZoneMap.class);
+        this.restApiHelper = new RestApiHelper<>(CurrentZoneMap.class);
     }
 
     public Response setCurrentZoneMapAsResponse(String zoneMapName){
-        return this.restApiHelper.post(null, "/configuration/v1/currentZoneMap/select/" + zoneMapName, target);
+        return this.restApiHelper.post(zoneMapName, "/configuration/v1/currentZoneMap/select/", target);
```
Comment (danshiovitz) [inline src/main/java/com/impinj/itemsense/client/coordinator/currentZoneMap/CurrentZoneMapController.java]: Is this really right? It seems weird the url now ends in a trailing slash

Comment (danshiovitz) [inline src/main/java/com/impinj/itemsense/client/coordinator/currentZoneMap/CurrentZoneMapController.java]: Oh, I see, the post method is overloaded. I feel like this is likely to lead to some confusion in usage, if there are two three-argument versions of post, and one means "add this id to the url, and there's no body" and the other is "use the url as-is, and here's the body".

Comment (davidyscott) [inline src/main/java/com/impinj/itemsense/client/coordinator/currentZoneMap/CurrentZoneMapController.java]: 100% agree, I did this to the POST method to follow the pattern set by the GET method
How would you feel if I refactored these method signatures to instead be
```
public Response post(WebTarget target, String... pathFragments)
public Response post(T entity, WebTarget target, String... pathFragments)

public Response put(WebTarget target, String... pathFragments)
public Response put(T entity, WebTarget target, String... pathFragments)

public Response get(WebTarget target, String... pathFragments) (this combines two existing methods)
public Response get(Map<String, Object> queryParams, WebTarget target, String... pathFragments)
```

**Comment by natke** [comment]
Approved.

**Comment by natke** [comment]
Looks good David!

**Pull request merged**